tests/workflows/protocols/test_pdos.py

```diff
@@ -0,0 +1,91 @@
+# -*- coding: utf-8 -*-
+"""Tests for the ``PwBandsWorkChain.get_builder_from_protocol`` method."""
+import pytest
+
+from aiida.engine import ProcessBuilder
+from aiida.plugins import WorkflowFactory
+
+from aiida_quantumespresso.common.types import ElectronicType, SpinType
+
+PdosWorkChain = WorkflowFactory('quantumespresso.pdos')
+
+
+def test_get_available_protocols():
+    """Test ``PdosWorkChain.get_available_protocols``."""
+    protocols = PdosWorkChain.get_available_protocols()
+    assert sorted(protocols.keys()) == ['fast', 'moderate', 'precise']
+    assert all('description' in protocol for protocol in protocols.values())
+
+
+def test_get_default_protocol():
+    """Test ``PdosWorkChain.get_default_protocol``."""
+    assert PdosWorkChain.get_default_protocol() == 'moderate'
+
+
+def test_default(fixture_code, generate_structure, data_regression, serialize_builder):
+    """Test ``PdosWorkChain.get_builder_from_protocol`` for the default protocol."""
+    pw_code = fixture_code('quantumespresso.pw')
+    dos_code = fixture_code('quantumespresso.dos')
+    projwfc_code = fixture_code('quantumespresso.projwfc')
+    structure = generate_structure()
+    builder = PdosWorkChain.get_builder_from_protocol(
+        pw_code=pw_code, dos_code=dos_code, projwfc_code=projwfc_code, structure=structure
+    )
+
+    assert isinstance(builder, ProcessBuilder)
+    data_regression.check(serialize_builder(builder))
+
+
+def test_electronic_type(fixture_code, generate_structure):
+    """Test ``PdosWorkChain.get_builder_from_protocol`` with ``electronic_type`` keyword."""
+    pw_code = fixture_code('quantumespresso.pw')
+    dos_code = fixture_code('quantumespresso.dos')
+    projwfc_code = fixture_code('quantumespresso.projwfc')
+    structure = generate_structure()
```

Comment:
You use this multiple times. Would be good to make a fixture `get_generator_inputs` that returns this as a dictionary. Then you just need to update one input each time (`electronic_type` in this case) and then you `**` it to `get_builder_from_protocol`.